src/Orleans/Streams/Predicates/FilterPredicateWrapperData.cs

```diff
@@ -50,6 +50,9 @@ internal class FilterPredicateWrapperData : IStreamFilterPredicateWrapper
         [NonSerialized]
         private StreamFilterPredicate predicateFunc;
 
+        // constructor used by serializator
+        private FilterPredicateWrapperData() { }
```

Comment:
If there is nothing that we can do about this on CoreCLR, then we have to accept that this restriction exists. @gabikliot what say you? I'm referring also to [this comment](https://github.com/dotnet/orleans/pull/1122#discussion_r47581713)